File: 백준/Bronze/2501. 약수 구하기/약수 구하기.py
answer = []
N, k = map(int,input().split())

for i in range(1,N+1):
    if N % i == 0:
        answer.append(i)

answer.sort() # 이미 값은 sort되어있으므로 굳이 할 필요 없다. 

if len(answer) < k: # len(answer)를 변수로 지정해도 된다.
    result = 0  # 결과 값을 바로 print 해도 된다.
else:
    result = answer[k-1]

print(result)

File: 백준/Bronze/2501. 약수 구하기/약수 구하기.cs
string[] input = Console.ReadLine().Split();
int N = int.Parse(input[0]);
int k = int.Parse(input[1]);

List<int> divisors = new List<int>();

for (int i = 1; i <= N; i++)
{
    if (N % i == 0)
    {
        divisors.Add(i);
    }
}

int result = (divisors.Count < k) ? 0 : divisors[k - 1];
Console.WriteLine(result);